gamemodes/terrortown/entities/weapons/weapon_bod_bodysnatch.lua

```diff
@@ -218,24 +240,20 @@ if CLIENT then
     end)
 
     -- If the player has snatched another player's name, show that name to other, non-allied, players
-    hook.Add("TTTTargetIDPlayerName", "Bodysnatcher_TTTTargetIDPlayerName", function(ply, cli, text, clr)
+    AddHook("TTTTargetIDPlayerName", "Bodysnatcher_TTTTargetIDPlayerName", function(ply, cli, text, clr)
         local disguiseName = ply:GetNWString("TTTBodysnatcherName", nil)
         if not disguiseName or #disguiseName == 0 then return end
 
-        -- Don't override the name for the player or their allies
-        if ply == cli then return end
-        if cli:IsSameTeam(ply) then return end
-
-        -- Show the overwritten name alongside their real name for allies
-        if ply == cli or cli:IsSameTeam(ply) then
+        -- Show the overwritten name alongside their real name for non-innocent allies
+        if not cli:IsInnocentTeam() and (ply == cli or cli:IsSameTeam(ply)) then
```

Comment:
I think this should be slightly rearranged so that innocents still see their own real name. It shouldn't ever matter for target ID but for chat it definitely should so I think its worth changing both for consistency.
```suggestion
        if ply == cli or (not cli:IsInnocentTeam() and cli:IsSameTeam(ply)) then
```